core/imds/src/main/java/software/amazon/awssdk/imds/internal/BaseEc2MetadataClient.java

```diff
@@ -29,21 +30,21 @@
 import software.amazon.awssdk.imds.EndpointMode;
 import software.amazon.awssdk.utils.AttributeMap;
 import software.amazon.awssdk.utils.IoUtils;
+import software.amazon.awssdk.utils.Lazy;
 import software.amazon.awssdk.utils.Logger;
+import software.amazon.awssdk.utils.NumericUtils;
 import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.internal.SystemSettingUtils;
 
 @SdkInternalApi
 public abstract class BaseEc2MetadataClient {
 
     protected static final Duration DEFAULT_TOKEN_TTL = Duration.of(21_600, ChronoUnit.SECONDS);
-    protected static final AttributeMap IMDS_HTTP_DEFAULTS =
-        AttributeMap.builder()
-                    .put(SdkHttpConfigurationOption.CONNECTION_TIMEOUT, Duration.ofSeconds(1))
-                    .put(SdkHttpConfigurationOption.READ_TIMEOUT, Duration.ofSeconds(1))
-                    .build();
 
     private static final Logger log = Logger.loggerFor(BaseEc2MetadataClient.class);
 
+    protected final Lazy<Duration> metadataServiceDuration = new Lazy<>(this::resolveMetadataServiceTimeoutDuration);
```

Comment:
@zoewangg 
Will update Ec2MetadataEndpointProvider to have this config since Ec2MetadataEndpointProvider has the respolveProfilFile method and this serviceTimeOut is related to Endpoint
In future is more settings then we can name Ec2MetadataEndpointProvider as Ec2MetadataConfigProvider.
Working on this changes , let me know any thoughts